giskard/utils/worker_pool.py

```diff
@@ -74,16 +70,21 @@ def __init__(self) -> None:
 
 @dataclass(frozen=True)
 class TimeoutData:
-    id: str
+    id: UUID
     end_time: float
 
 
-@dataclass(frozen=True)
 class GiskardTask:
-    callable: Callable
+    fn: Callable
     args: Any
     kwargs: Any
-    id: str = field(default_factory=_generate_task_id)
+    job_id: UUID
+
+    def __init__(self, job_id: UUID, fn: Callable, args: Any, kwargs: Any):
```

Comment:
This should be remove, since dataclass will define an implicit constructor.
I guess job_id should be moved as first attribute to respect order.

Reply:
done